community/kernel/src/main/java/org/neo4j/helpers/SocketAddress.java

```diff
@@ -58,7 +58,14 @@ public InetSocketAddress socketAddress()
     @Override
     public String toString()
     {
-        return hostname + ":" + port;
+        if (hostname.contains(":"))
```

Comment:
Would this `#toString()` implementation return strange string when given host name already contains `[...]` like `[::1]`?
Maybe is it worth doing smth like:
```
InetAddress address = InetAddress.getByName( hostname );
if ( address instanceof Inet6Address )
{
    return "[" + address.getHostName() + "]:" + port;
}
else
{
    return address.getHostName() + ":" + port;
}
```